java-checks/src/test/files/checks/security/CryptographicKeySizeCheck.java

```diff
@@ -26,8 +32,51 @@ public void report_twice() throws NoSuchAlgorithmException {
     keyGen.initialize(1024); // Noncompliant
     keyGen.initialize(1023); // Noncompliant
   }
+
+  public void report_only_once(int size) throws NoSuchAlgorithmException {
+    if (size == 1) {
+      KeyPairGenerator keyGen = KeyPairGenerator.getInstance("DH");
+      keyGen.initialize(1); // Noncompliant {{Use a key length of at least 2048 bits for DH cipher algorithm.}}
+    } else if(size == 2) {
+      KeyPairGenerator keyGen = KeyPairGenerator.getInstance("RSA");
+      keyGen.initialize(2); // Noncompliant {{Use a key length of at least 2048 bits for RSA cipher algorithm.}}
+    } else if(size == 3) {
+      KeyGenerator keyGen = KeyGenerator.getInstance("AES");
+      keyGen.init(3); // Noncompliant {{Use a key length of at least 128 bits for AES cipher algorithm.}}
+    } else {
+      KeyPairGenerator keyGen = KeyPairGenerator.getInstance("DH");
+      keyGen.initialize(2048); // Compliant
+    }
+  }
+
+  public void not_assigned() throws NoSuchAlgorithmException {
+    KeyPairGenerator.getInstance("RSA"); // Not assigned, nothing to do
+  }
+}
+
+class RSA2 {
+  KeyPairGenerator keyGen = KeyPairGenerator.getInstance("RSA");
+
+  RSA2() throws NoSuchAlgorithmException {}
+
+  public void key_variable_this() throws NoSuchAlgorithmException {
+    KeyPairGenerator keyGen = KeyPairGenerator.getInstance("RSA");
+    this.keyGen.initialize(1024); // Compliant, only support when declared in the same method
+  }
+
+  public void key_variable_instance() throws NoSuchAlgorithmException {
+    RSA2 rsa2 = new RSA2();
+    rsa2.keyGen = KeyPairGenerator.getInstance("RSA");
+    rsa2.setKeyGen();
+    rsa2.keyGen.initialize(1024); // Compliant, only support when declared in the same method
```

Comment:
I think it's better to not report member select, I expect this pattern to be super unusual, and FP prone if we don't dive deeply into the subject...

We will have FN in the case where `setKeyGen()` is not called, but I think it's acceptable.

Let me know what you think!

Reply:
Fair point, LGTM!